### PR TITLE
3074 TDP Datafile page for DIGIT and Sys Admin groups

### LINKED ISSUE #3074: TDP Data Files page permissions for DIGIT & Sys Admin user groups
**Description:**
Currently users assigned to the DIGIT or System Admin user groups cannot reach and browse TDP's Data Files page. This ticket adds those permissions for both groups. Functionality should mirror how OFA Admin usergroup users can currently pick from STTs on the data files page on the frontend of the app (see screenshot below). 

**Acceptance Criteria:** 
- [x] DIGIT user group has permissions for accessing TDP frontend Data Files pages
- [ ] System Admin user group has permissions for accessing TDP frontend Data Files pages
- [ ] Testing Checklist has been run and all tests pass
- [ ] README is updated, if necessary

**Tasks:**
- [x] Implement permission group changes
- [ ] Confirm that the data files page can be browsed by both groups & submission history access is included
- [ ] Run Testing Checklist and confirm all tests pass

**Supporting Documentation:**
[Research Synthesis w/ all DAC issues](https://app.gitbook.com/o/Yd4Wv0Fi89kSKpmakWrn/s/uAlDVLhlFIbJw6huUS2x/research-drafts/ofa-admin-experience-synthesis)

image

### PR DESCRIPTION
**Pull request by raftmsohani**
## Summary of Changes
Pull request closes #3074 _

## How to Test
These steps are generic, please adjust as necessary.
```
cd tdrs-frontend && docker-compose -f docker-compose.yml -f docker-compose.local.yml up -d
cd tdrs-backend && docker-compose -f docker-compose.yml -f docker-compose.local.yml up -d 
```

1. Open http://localhost:3000/ and sign in.
1. Proceed with functional tests as described herein.
1. Test steps should be captured in the demo GIF(s) and/or screenshots below.
> *Demo GIF(s) and screenshots for testing procedure*

## Deliverables
_More details on how deliverables herein are assessed included [here](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverables)._

### [Deliverable 1: Accepted Features](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-1-Accepted-Features)

Checklist of ACs:
- [x] DIGIT user group has permissions for accessing TDP frontend Data Files pages
- [x] System Admin user group has permissions for accessing TDP frontend Data Files pages
- [x] Testing Checklist has been run and all tests pass
+ [x] **`lfrohlich`** and/or **`adpennington`**  confirmed that ACs are met.

### [Deliverable 2: Tested Code](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-2-Tested-Code)

+ Are all areas of code introduced in this PR meaningfully tested?
  + [x] If this PR introduces backend code changes, are they meaningfully tested?
  + [x] If this PR introduces frontend code changes, are they meaningfully tested?
+ Are code coverage minimums met?
  + [x] Frontend coverage: [_insert coverage %_] (see `CodeCov Report` comment in PR)
  + [x] Backend coverage: [_insert coverage %_] (see `CodeCov Report` comment in PR)

### [Deliverable 3: Properly Styled Code](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-3-Properly-Styled-Code)

+ [x] Are backend code style checks passing on CircleCI?
+ [x] Are frontend code style checks passing on CircleCI?
+ [x] Are code maintainability principles being followed?

### [Deliverable 4: Accessible](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-4-Accessibility)

+ [ ] Does this PR complete the epic? 
+ [ ] Are links included to any other gov-approved PRs associated with epic?
+ [ ] Does PR include documentation for Raft's a11y review? 
+ [ ] Did automated and manual testing with `iamjolly` and `ttran-hub` using Accessibility Insights reveal any errors introduced in this PR?

### [Deliverable 5: Deployed](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-5-Deployed)

+ [x] Was the code successfully deployed via automated CircleCI process to development on Cloud.gov?

### [Deliverable 6: Documented](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-6-Code-documentation)

+ [x] Does this PR provide background for why coding decisions were made?
+ [ ] If this PR introduces backend code, is that code easy to understand and sufficiently documented, both inline and overall?
+ [ ] If this PR introduces frontend code, is that code easy to understand and sufficiently documented, both inline and overall?
+ [ ] If this PR introduces dependencies, are their licenses documented?
+ [x] Can reviewer explain and take ownership of these elements presented in this code review?

### [Deliverable 7: Secure](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-7-Secure)

+ [x] Does the OWASP Scan pass on CircleCI?
+ [ ] Do manual code review and manual testing detect any new security issues?
+ [ ] If new issues detected, is investigation and/or remediation plan documented? 

### [Deliverable 8: User Research](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-8-User-Research)

Research product(s) clearly articulate(s):
+ [ ] the purpose of the research
+ [ ] methods used to conduct the research 
+ [ ] who participated in the research
+ [ ] what was tested and how
+ [ ] impact of research on TDP
+ [ ] (_if applicable_) final design mockups produced for TDP development

### CODE REVIEW

File: tdrs-backend/tdpservice/users/migrations/0041_users_digit_group_add_datafile_permission.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+
+    datafile_permissions = get_permission_ids_for_model(
+        'data_files',
+        'datafile',
+        filters=[view_permissions_q, add_permissions_q]
+    )
+
+    # Assign model permissions
+    digit.permissions.add(*datafile_permissions, *stt_permissions)
```
Comment (jtimpe) [inline tdrs-backend/tdpservice/users/migrations/0041_users_digit_group_add_datafile_permission.py:32]: was it necessary to add object-level permissions? i don't recall ever using these, we normally just use the group/role assignments

**Review by andrew-jameson — APPROVED** [review]
LGTM

**Review by elipe17 — APPROVED** [review]
LGTM 🤘

File: tdrs-backend/tdpservice/users/migrations/0041_users_digit_group_add_datafile_permission.py (hunk 1 above)
Comment (raftmsohani) [reply tdrs-backend/tdpservice/users/migrations/0041_users_digit_group_add_datafile_permission.py:32]: Not really, but I still decided to do it to match permissions with frontend. I think we should have let the backend manage the permission as whole, but I guess this needs some rework.

**Comment by ADPennington** [comment]
@klinkoberstar  im reviewing this PR :) @andrew-jameson if possible, can we add her to this repo?

**Review by ADPennington — APPROVED** [review]
lgtm @raftmsohani 🚀 my testing notes are below cc: @klinkoberstar 

---
_**current state (tested in staging environment)**_

1. `Sys admin` group can see and access data files page but do not have the ability see dropdown menus to view submission history because STT not set in user profile

Sys_before


2. `DIGIT` group cannot see and access data files page (exception: site map shows that this page exists but user cannot access it :warning:)
digit_before



3. `DIGIT` group can access the following pages in DAC

- data files
- search indexes
- parsers
digit_before_dac


_**this branch**_

4. `Sys admin` group can see and access data files page, can submit data, and can view submission history

Sys_after
Sys_after_submitdata
Sys_after_history

5. `DIGIT` group can see and access data files page, can submit data, and can view submission history, and can access all needed pages in DAC (plus STTs page)

digit_after
digit_after_dac

**Pull request merged**